src/Common/src/Interop/Linux/System.Net.Security.Native/Interop.HeimdalNtlm.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class HeimdalNtlm
+    {
+        internal static byte[] CreateNegotiateMessage(uint flags)
+        {
+            SafeNtlmBufferHandle data;
+            int dataLen;
+            int status = NetSecurity.HeimNtlmEncodeType1(flags, out data, out dataLen);
+            NetSecurity.HeimdalNtlmException.AssertOrThrowIfError("HeimNtlmEncodeType1 failed", status);
```

Comment:
Should this be moved to inside of the using so that the SafeNtlmBufferHandle is disposed even if the operation fails?  Or is data guaranteed to be null in the event of a failure? (If it's not-null but IsInvalid, we should still dispose of it.)